src/dataServices/dynamoDbBundleServiceHelper.ts

```diff
@@ -253,4 +256,231 @@ export default class DynamoDbBundleServiceHelper {
+        dynamoDbHelper: DynamoDbHelper,
+        tenantId?: string,
+    ) {
+        console.log(requests);
+        const deleteRequests: any = [];
+        const createRequests: any = [];
+        const updateRequests: any = [];
+
+        const batchReadWriteResponses: BatchReadWriteResponse[] = [];
+        requests.forEach(async (request) => {
+            let vid = 0;
+            let { id } = request;
+            const { resourceType, operation } = request;
+            let item;
+            // we need to query to get the VersionID of the resource for non-create operations
+            if (operation === 'create') {
+                vid = 1;
+                id = request.id ? request.id : uuidv4();
+            } else {
+                try {
+                    item = await dynamoDbHelper.getMostRecentUserReadableResource(resourceType, id, tenantId);
+                    console.log('retrieved item:', item);
+                    vid = Number(item.resource?.meta.versionId);
+                } catch (e: any) {
+                    console.log(`Failed to find resource ${id}`);
+                    batchReadWriteResponses.push({
+                        id,
+                        vid: '',
+                        operation,
+                        resourceType,
+                        resource: {},
+                        lastModified: '',
+                    });
+                    return;
+                }
+            }
+            switch (operation) {
+                case 'create': {
+                    item = DynamoDbUtil.prepItemForDdbInsert(
+                        request.resource,
+                        id,
+                        vid,
+                        DOCUMENT_STATUS.AVAILABLE,
+                        tenantId,
+                    );
+
+                    createRequests.push({
+                        PutRequest: {
+                            Item: DynamoDBConverter.marshall(item),
+                        },
+                    });
+
+                    batchReadWriteResponses.push({
+                        id,
+                        vid: item.meta.versionId,
+                        operation: request.operation,
+                        lastModified: item.meta.lastUpdated,
+                        resourceType: request.resource.resourceType,
+                        resource: request.resource,
+                    });
+                    break;
+                }
+                case 'update': {
+                    // increment the vid
+                    vid += 1;
+                    item = DynamoDbUtil.prepItemForDdbInsert(
+                        { ...item?.resource, ...request.resource },
+                        id,
+                        vid,
+                        DOCUMENT_STATUS.AVAILABLE,
+                        tenantId,
+                    );
+                    // we need to delete the old verison, and create a new version
+                    updateRequests.push([
```

Comment:
What happens if the update on old resource goes through and the new resource fail to create?